DiceKit/FrequencyDistribution.swift

```diff
@@ -222,9 +239,25 @@ extension FrequencyDistribution {
         }
     }
     
-    // TODO: public func divide(x: FrequencyDistribution) -> FrequencyDistribution
-    // Needed to support power(-n)
-    
+    public func divide(y: FrequencyDistribution) -> FrequencyDistribution {
+        guard let initialK = orderedOutcomes.first, lastK = orderedOutcomes.last else {
+            return .additiveIdentity
+        }
+        guard let firstY = y.orderedOutcomes.first, lastY = y.orderedOutcomes.last, firstYFrequency = y[firstY] where firstYFrequency != 0.0 else {
+            fatalError("Divide by zero! :(")
+        }
+        var xFrequencies: FrequenciesPerOutcome = [:]
+        var p: Frequency
+        for k:Outcome in initialK...(lastK + lastY) {
+            p = 0.0
+            for (n, frequency) in xFrequencies {
+                p += frequency * (y[k - n] ?? 0)
+            }
+            xFrequencies[k - firstY] = ((frequenciesPerOutcome[k] ?? 0) - p) / firstYFrequency
+        }
+        let delta = ProbabilityMassConfig.probabilityEqualityDelta
+        return FrequencyDistribution(xFrequencies).filterZeroFrequencies(delta)
```

Comment:
`filterZeroFrequencies` Is the right public API. I don't want to expose a weird init, to the public it should *always* filter the zeroes. We are almost there now, just change the init to take the delta, and make the method call the init with the default.